Fix `docker/linux-image.sh` script

gamma0987 opened the pull request:
As promised here's the fixed `linux-image.sh` script.

As outlined in #1399 this pr fixes:

* Missing gpg key 8D69674688B6CB36 (Debian Ports Archive Automatic Signing Key (2024) <ftpmaster@ports-master.debian.org>)
* The binaries and busybox might also be installed into `/usr/bin/` instead of `/bin`.
* The kernel modules might also installed into `/usr/lib/modules` instead of `/lib/modules`.
* The kernel modules might be compressed with `xz` and end with `*.ko.xz` instead of just `*.ko`

The first commit adds some vim special files to `.gitignore` and in the second commit the script was formatted with `shfmt` with standard settings and some minor issues from `shellcheck` were fixed.

The third commit fixes the actual issues. Not sure if you wanna keep the first two commits but they made working on this file easier.

Closes #1399

Emilgardis commented:
/ci try

Emilgardis commented:
ugh, forgot I hadn't fixed the try action yet, it skips the actual build of the images :D

gamma0987 commented:
I was just about to ask if the script is actually running somewhere :)

Emilgardis commented:
/ci try

should be fixed now!

Emilgardis commented:
The `mips*-musl` errors are unrelated, should be fixed with https://github.com/cross-rs/cross/pull/1407

You can see the latest CI run here https://github.com/cross-rs/cross/actions/runs/7391006735

As seen, the ci still fails, but that's because it's during [`qemu-user`](https://github.com/cross-rs/cross/actions/runs/7391006735/job/20107107972), do you think it would make sense to remove qemu-user from powerpc? That would be done [here](https://github.com/cross-rs/cross/blob/4da645bc4553fbe6b16df320d7c5c9cc6839dba8/targets.toml#L181)

gamma0987 commented:
> basically there's no changes

That's because the ci job didn't pick up the commits and the new `linux-image.sh` script. I'm not sure what's going wrong but I downloaded the logs of https://github.com/cross-rs/cross/actions/runs/7438545066?pr=1405 and did a grep:

```shell
❯ rg 'curl.*2023' .
./try  target (powerpc64-unknown-linux-gnu,ubuntu-latest)/10_Build Docker image.txt
7950:2024-01-07T13:49:01.9909003Z #22 10.43 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./try  target (riscv64gc-unknown-linux-gnu,ubuntu-latest)/10_Build Docker image.txt
7633:2024-01-07T13:49:07.4924205Z #22 11.40 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./20_try  target (powerpc64-unknown-linux-gnu,ubuntu-latest).txt
9265:2024-01-07T13:49:01.9909044Z #22 10.43 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./4_try  target (x86_64-unknown-linux-gnucentos,ubuntu-latest).txt
2457:2024-01-07T13:49:00.4962792Z #14 17.02 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./try  target (aarch64-unknown-linux-gnucentos,ubuntu-latest)/10_Build Docker image.txt
1185:2024-01-07T13:37:26.7686432Z #15 19.64 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./62_try  target (aarch64-unknown-linux-gnucentos,ubuntu-lates.txt
2502:2024-01-07T13:37:26.7686467Z #15 19.64 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./24_try  target (sparc64-unknown-linux-gnu,ubuntu-latest).txt
8965:2024-01-07T13:49:16.5505277Z #22 11.54 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./try  target (x86_64-unknown-linux-gnucentos,ubuntu-latest)/10_Build Docker image.txt
1140:2024-01-07T13:49:00.4962754Z #14 17.02 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./try  target (sparc64-unknown-linux-gnu,ubuntu-latest)/10_Build Docker image.txt
7648:2024-01-07T13:49:16.5505246Z #22 11.54 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./try  target (powerpc-unknown-linux-gnu,ubuntu-latest)/10_Build Docker image.txt
7811:2024-01-07T13:49:18.8106413Z #22 10.77 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./19_try  target (powerpc-unknown-linux-gnu,ubuntu-latest).txt
9122:2024-01-07T13:49:18.8106437Z #22 10.77 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O

./22_try  target (riscv64gc-unknown-linux-gnu,ubuntu-latest).txt
8945:2024-01-07T13:49:07.4924240Z #22 11.40 + curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023}.key' -O
```

If this would have been the new `linux-image.sh` script this line should include the `2024` key:

```
curl --retry 3 -sSfL 'https://www.ports.debian.org/archive_{2020,2021,2022,2023,2024}.key' -O
```

gamma0987 commented:
Looks like the error already happens in the `actions/checkout@v3` step

```
0s
Run actions/checkout@v3
Syncing repository: cross-rs/cross
Getting Git version info
Temporarily overriding HOME='/home/runner/work/_temp/9e0762cf-65f4-49d3-835d-a4d525697544' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
/usr/bin/git config --global --add safe.directory /home/runner/work/cross/cross
Deleting the contents of '/home/runner/work/cross/cross'
Initializing the repository
Disabling automatic garbage collection
Setting up auth
Fetching the repository
Determining the checkout info
Checking out the ref
  /usr/bin/git checkout --progress --force -B main refs/remotes/origin/main
  Switched to a new branch 'main'
  branch 'main' set up to track 'origin/main'.
/usr/bin/git log -[1](https://github.com/cross-rs/cross/actions/runs/7438545066/job/20237413186?pr=1405#step:3:1) --format='%H'
'4da645bc4553fbe6b16df320d7c5c9cc6839dba8'
```

because it checks out the main branch instead of the pull request.

I've rebased onto the `main` branch to pick up the latest ci changes from #1406. Maybe that helps...

Emilgardis commented:
hmm, thank you for looking into it @gamma0987, with that information I know what's wrong. I'll fix it

gamma0987 commented:
Sure. Thanks for fixing it

Emilgardis commented:
Now it *should* work!

/ci try

Emilgardis commented:
/ci try

gamma0987 commented:
Looks like it worked so far. Any idea why the `zig` job fails? The `riscv64gc-unknown-linux-gnu` target doesn't build, too. Since there's something wrong in the `linux-image.sh` script, I can try fixing it.

Emilgardis commented:
/ci try

Emilgardis commented:
Diff for [comment](comment)

- [Run](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20250946738)
- [✅ aarch64-linux-android](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251000708?pr=1405#step:10:1)
- [✅ aarch64-unknown-linux-gnu:centos](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251000895?pr=1405#step:10:1)
- [✅ aarch64-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251001051?pr=1405#step:10:1)
- [✅ aarch64-unknown-linux-musl](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251001192?pr=1405#step:10:1)
- [✅ aarch64-unknown-freebsd](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251001365?pr=1405#step:10:1)
- [✅ arm-linux-androideabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251001549?pr=1405#step:10:1)
- [✅ arm-unknown-linux-gnueabihf](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251001751?pr=1405#step:10:1)
- [✅ arm-unknown-linux-gnueabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251001928?pr=1405#step:10:1)
- [✅ arm-unknown-linux-musleabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251002062?pr=1405#step:10:1)
- [✅ arm-unknown-linux-musleabihf](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251002218?pr=1405#step:10:1)
- [✅ armv5te-unknown-linux-gnueabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251002357?pr=1405#step:10:1)
- [✅ armv5te-unknown-linux-musleabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251002491?pr=1405#step:10:1)
- [✅ armv7-linux-androideabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251002613?pr=1405#step:10:1)
- [✅ armv7-unknown-linux-gnueabihf](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251002735?pr=1405#step:10:1)
- [✅ armv7-unknown-linux-gnueabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251002872?pr=1405#step:10:1)
- [✅ armv7-unknown-linux-musleabihf](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251003023?pr=1405#step:10:1)
- [✅ armv7-unknown-linux-musleabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251003180?pr=1405#step:10:1)
- [✅ i586-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251003309?pr=1405#step:10:1)
- [✅ cross](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251003434?pr=1405#step:10:1)
- [✅ i586-unknown-linux-musl](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251003597?pr=1405#step:10:1)
- [✅ i686-linux-android](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251003737?pr=1405#step:10:1)
- [✅ i686-pc-windows-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251003867?pr=1405#step:10:1)
- [✅ i686-unknown-freebsd](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251004013?pr=1405#step:10:1)
- [✅ i686-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251004154?pr=1405#step:10:1)
- [✅ i686-unknown-linux-musl](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251004311?pr=1405#step:10:1)
- [✅ mips64el-unknown-linux-gnuabi64](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251004459?pr=1405#step:10:1)
- [❌ mips64el-unknown-linux-muslabi64](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251004584?pr=1405#step:10:1)
- [✅ mips64-unknown-linux-gnuabi64](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251004724?pr=1405#step:10:1)
- [❌ mips64-unknown-linux-muslabi64](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251004867?pr=1405#step:10:1)
- [❌ mipsel-unknown-linux-musl](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005000?pr=1405#step:10:1)
- [✅ mipsel-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005114?pr=1405#step:10:1)
- [✅ mips-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005228?pr=1405#step:10:1)
- [❌ mips-unknown-linux-musl](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005359?pr=1405#step:10:1)
- [✅ powerpc64le-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005493?pr=1405#step:10:1)
- [✅ powerpc64-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005594?pr=1405#step:10:1)
- [✅ powerpc-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005710?pr=1405#step:10:1)
- [❌ riscv64gc-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005844?pr=1405#step:10:1)
- [✅ s390x-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251005960?pr=1405#step:10:1)
- [✅ sparc64-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006075?pr=1405#step:10:1)
- [❌ sparcv9-sun-solaris](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006215?pr=1405#step:10:1)
- [✅ thumbv6m-none-eabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006320?pr=1405#step:10:1)
- [✅ thumbv7em-none-eabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006450?pr=1405#step:10:1)
- [✅ thumbv7em-none-eabihf](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006557?pr=1405#step:10:1)
- [✅ thumbv7m-none-eabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006673?pr=1405#step:10:1)
- [✅ thumbv7neon-linux-androideabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006805?pr=1405#step:10:1)
- [✅ thumbv7neon-unknown-linux-gnueabihf](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251006956?pr=1405#step:10:1)
- [✅ thumbv8m.base-none-eabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251007073?pr=1405#step:10:1)
- [✅ thumbv8m.main-none-eabi](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251007247?pr=1405#step:10:1)
- [✅ thumbv8m.main-none-eabihf](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251007388?pr=1405#step:10:1)
- [✅ wasm32-unknown-emscripten](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251007523?pr=1405#step:10:1)
- [✅ x86_64-apple-darwin](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251007666?pr=1405#step:10:1)
- [✅ x86_64-linux-android](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251007800?pr=1405#step:10:1)
- [✅ x86_64-pc-windows-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251007989?pr=1405#step:10:1)
- [✅ x86_64-pc-windows-msvc](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251008136?pr=1405#step:10:1)
- [❌ x86_64-sun-solaris](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251008325?pr=1405#step:10:1)
- [✅ x86_64-unknown-dragonfly](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251008474?pr=1405#step:10:1)
- [✅ x86_64-unknown-freebsd](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251008685?pr=1405#step:10:1)
- [✅ x86_64-unknown-illumos](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251008851?pr=1405#step:10:1)
- [✅ x86_64-unknown-linux-gnu:centos](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251009015?pr=1405#step:10:1)
- [✅ x86_64-unknown-linux-gnu](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251009157?pr=1405#step:10:1)
- [✅ x86_64-unknown-linux-musl](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251009282?pr=1405#step:10:1)
- [✅ x86_64-unknown-netbsd](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251009414?pr=1405#step:10:1)
- [❌ zig](https://github.com/cross-rs/cross/actions/runs/7444481056/job/20251009535?pr=1405#step:10:1)

Emilgardis commented:
Oops, trying out different ways to represent the data and forgot to remove the command that actually makes the comment :D

Emilgardis commented:
I think we can keep the commit anyway, we'll make fixes for the remaining targets

Emilgardis reviewed (approved):
Thank you!

gamma0987 commented:
Sure :)
